File: inject_tuner.py
import re

with open('c:/Bloom/BLOOM_3.github.io/product.html', 'r', encoding='utf-8') as f:
    html = f.read()

# Add ID to Curation Preview text
idx = html.find('Curation Preview')
if idx != -1:
    tag_start = html.rfind('<p', 0, idx)
    if 'id=' not in html[tag_start:idx]:
        html = html[:tag_start] + html[tag_start:].replace('<p ', '<p id="curation-preview-text" ', 1)

# Inject the control panel just before </body>
control_panel = """
  <!-- TEMPORARY CONTROL PANEL FOR POSITION TUNING -->
  <div id="position-tuner" style="position: fixed; bottom: 20px; right: 20px; background: rgba(0,0,0,0.8); color: white; padding: 20px; border-radius: 10px; z-index: 9999; font-family: sans-serif; display: flex; flex-direction: column; gap: 10px;">
    <h3 style="margin: 0; font-size: 16px;">Layout Tuner</h3>
    
    <label>
      Texts X-Position (Left): <span id="val-texts-x">40</span>px<br>
      <input type="range" id="slider-texts-x" min="0" max="500" value="40" style="width: 250px;">
    </label>
    
    <label>
      Cards Y-Position (padding-top): <span id="val-cards-y">459</span>px<br>
      <input type="range" id="slider-cards-y" min="0" max="800" value="459" style="width: 250px;">
    </label>

    <label>
      Curation Preview Y-Position (Top): <span id="val-preview-y">393</span>px<br>
      <input type="range" id="slider-preview-y" min="0" max="800" value="393" style="width: 250px;">
    </label>
  </div>

  <script>
    document.getElementById('slider-texts-x').addEventListener('input', (e) => {
      const val = e.target.value;
      document.getElementById('val-texts-x').innerText = val;
      const t1 = document.getElementById('curation-text-1');
      const t2 = document.getElementById('curation-text-2');
      if (t1) t1.style.left = val + 'px';
      if (t2) t2.style.left = val + 'px';
    });

    document.getElementById('slider-cards-y').addEventListener('input', (e) => {
      const val = e.target.value;
      document.getElementById('val-cards-y').innerText = val;
      const container = document.getElementById('curation-cards-container');
      if (container) container.style.paddingTop = val + 'px';
    });

    document.getElementById('slider-preview-y').addEventListener('input', (e) => {
      const val = e.target.value;
      document.getElementById('val-preview-y').innerText = val;
      const preview = document.getElementById('curation-preview-text');
      if (preview) preview.style.top = val + 'px';
    });
  </script>
"""

html = html.replace('</body>', control_panel + '\n</body>')

with open('c:/Bloom/BLOOM_3.github.io/product.html', 'w', encoding='utf-8') as f:
    f.write(html)
print("Control panel injected.")

File: inject_tuner.js
const fs = require('fs');
let html = fs.readFileSync('product.html', 'utf8');

const widget = `
<div id="temp-tuner" style="position: fixed; bottom: 20px; right: 20px; z-index: 99999; background: rgba(255,255,255,0.9); padding: 20px; border: 1px solid #ccc; border-radius: 8px; box-shadow: 0 4px 12px rgba(0,0,0,0.2); font-family: sans-serif;">
  <p style="margin-bottom: 10px;"><strong>Card Position Tuner</strong></p>
  <label>Top: <input type="range" id="tuner-top" min="0" max="1000" value="330" oninput="document.querySelector('.hero-card-position-wrapper').style.top = this.value + 'px'; document.getElementById('top-val').innerText = this.value;"></label>
  <span id="top-val">330</span>px<br>
  <label style="margin-top:10px; display:block;">Left: <input type="range" id="tuner-left" min="0" max="1500" value="660" oninput="document.querySelector('.hero-card-position-wrapper').style.left = this.value + 'px'; document.getElementById('left-val').innerText = this.value;"></label>
  <span id="left-val">660</span>px
</div>
`;

if (!html.includes('temp-tuner')) {
    html = html.replace('</body>', widget + '</body>');
    fs.writeFileSync('product.html', html);
    console.log("Injected tuner");
} else {
    console.log("Tuner already exists");
}
